x-pack/plugins/lens/public/editor_frame_service/editor_frame/workspace_panel/workspace_panel.tsx

```diff
@@ -251,7 +256,9 @@ export function InnerWorkspacePanel({
   }
 
   function renderVisualization() {
-    if (expression === null) {
+    // we don't want to render the emptyWorkspace on visualizing field from Discover
+    // as it is specific for the drag and drop functionality and can confuse the users
+    if (expression === null && !visualizeTriggerFieldContext) {
```

Comment:
As `visualizeTriggerFieldContext` is not removed after it's "consumed" by `applyVisualizeFieldSuggestions`, the visualization will get stuck in a loading screen if you enter an invalid configuration after coming to Lens from Discover:
* Click on "Visualize" in Discover
* Remove y dimension
* Spinner keeps running

<img width="1472" alt="Screenshot 2020-10-02 at 13 23 32" src="https://user-images.githubusercontent.com/1508364/94918267-83642400-04b2-11eb-93fb-2c72cf0d9997.png">

In the EditorFrame component, could you put the `props.initialContext` into a local state
```ts
const [visualizeTriggerFieldContext, setVisualizeTriggerFieldContext] = useState(props.initialContext)
```

and reset it after using it for init:
```ts
  useEffect(() => {
    if (allLoaded && props.initialContext && !props.doc) {
      applyVisualizeFieldSuggestions({
        datasourceMap: props.datasourceMap,
        datasourceStates: state.datasourceStates,
        visualizationMap: props.visualizationMap,
        activeVisualizationId: state.visualization.activeId,
        visualizationState: state.visualization.state,
        visualizeTriggerFieldContext: props.initialContext,
        dispatch,
      });
      setVisualizeTriggerFieldContext(undefined);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [allLoaded]);
```

That should fix the bug.

Reply:
I missed that, thanx for this ❤️